src/main/java/chess/domain/piece/Piece.java

```diff
@@ -0,0 +1,86 @@
+package chess.domain.piece;
+
+import chess.domain.Position;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class Piece {
+
+    protected final Set<Direction> directions;
+    protected final Color color;
+
+    protected Piece(final Color color) {
+        directions = new HashSet<>();
+        this.color = color;
+    }
+
+    abstract public List<Position> findPath(final Position source, final Position target);
```

Comment:
```suggestion
    public abstract List<Position> findPath(final Position source, final Position target);
```
다른 부분도 접근제어자가 제일 앞으로 오도록 챙겨봐주세요